### PR TITLE
Add Farming Simulator 2022/2019

### PR DESCRIPTION
**Pull request by Vito0912**
Added Farming Simulator 2022. It may also work with older versions, although I currently only own Farming Simulator 19 and 17. I will need to investigate whether I can set up a server for them. As of now, this has only been tested with a Farming Simulator 2022 server.

Note:
Server queries are not possible. To obtain the necessary token, you must own or have permission to access the web interface.

I have not tested whether the `token` works when used in code; my testing was limited to using it with the command line.

### CODE REVIEW

**Comment by Vito0912** [comment]
I am not able to start a Farming Simulator 17 Server. It will likely work, but I cannot test it.

File: GAMES_LIST.md [hunk 1]
```diff
@@ -460,6 +461,10 @@ Conan Exiles never responds to player query.
 ### <a name="minecraft">Minecraft
 Many Minecraft servers do not respond with players data.
 
+### <a name='farmingsimulator'>Farming Simulator
+Farming Simulator 2022 servers need a token (reffered as code in the game). It can be obtained at your server's web interface (http://ip:port/settings.html). It can be passed to GameDig with the additional option: `token`
```
Comment (CosminPerRam) [inline GAMES_LIST.md:465]: `Farming Simulator 2022 servers need a token`, so, only FS22 or both? as this note is added on both of them.

File: lib/QueryRunner.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default class QueryRunner {
         errors.push(e)
       }
     }
-
+
```
Comment (CosminPerRam) [inline lib/QueryRunner.js:105]: This line doesnt need to be changed.

File: lib/games.js [hunk 3]
```diff
@@ -1186,6 +1186,22 @@ export const games = {
       protocol: 'ase'
     }
   },
+  farmingsimulator2019: {
+    name: 'Farming Simulator 2019',
+    release_year: 2021,
```
Comment (CosminPerRam) [inline lib/games.js:1191]: Should be 2018.

**Review by CosminPerRam — CHANGES_REQUESTED** [review]

**Comment by CosminPerRam** [comment]
As the queries require tokens to be made, you think you could join the Discord server or message me on Discord regarding providing me the needed ones? As I dont own the game I cannot really verify that this PR is valid (not that I don't trust you, but its just to make sure all is good).

**Comment by Vito0912** [comment]
> So as the changes in this PR, we would need to also test FS19.

I tested it with Farming Simulator 19

File: protocols/farmingsimulator.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          name: $(this).text(),
+          raw: {
+            isAdmin: $(this).attr('isAdmin') === 'true',
+            uptime: parseInt($(this).attr('uptime'), 10)
+          }
+        });
+      }
+    });
+
+    state.mods = [];
```
Comment (CosminPerRam) [inline protocols/farmingsimulator.js:38]: `mods` is not a stabilized field, so it should be under `state.raw`.

File: GAMES_LIST.md [hunk 5]
```diff
@@ -122,6 +122,8 @@
 | factorio             | Factorio                                         |                                                 |
 | farcry               | Far Cry                                          |                                                 |
 | farcry2              | Far Cry 2                                        |                                                 |
+| farmingsimulator2019 | Farming Simulator 2019                           | [Notes](#farmingsimulator)                      |
```
Comment (CosminPerRam) [inline GAMES_LIST.md:125]: The name of the game should be `Farming Simulator 19`, same for 22.

**Review by CosminPerRam — CHANGES_REQUESTED** [review]
Privately validated that the queries work.
Please also run eslint (:

File: GAMES_LIST.md (hunk 5 above)
Comment (Vito0912) [reply GAMES_LIST.md:125]: Should the naming of the key also be changed?

Comment (CosminPerRam) [reply GAMES_LIST.md:125]: Yep.

Comment (CosminPerRam) [reply GAMES_LIST.md:125]: (Requesting the name change as seeing on their website that they call the game "Farming Simulator 22", not "Farming Simulator 2022")

Comment (CosminPerRam) [reply GAMES_LIST.md]: (same goes for the names in the `games.js` file).

**Comment by podrivo** [comment]
@Vito0912 Sorry, but still a conflict to resolve before merging...

**Review by CosminPerRam — APPROVED** [review]
Good to merge, but please also run `eslint` of the protocol file.

**Comment by podrivo** [comment]
@Vito0912 Do we have a `version` field in the query response? Wondering if we should add a `version` top level for Farming Simulator as well.

**Comment by Vito0912** [comment]
@podrivo Yeah, I am pretty sure it does have a version in it's response. Can't remember that there was a top level `version`. I will add it as soon as I have time.

**Comment by podrivo** [comment]
No sweat!
If you could add a query response at the original post, I'll make sure to add it in the other PR (https://github.com/gamedig/node-gamedig/pull/532).

**Comment by Vito0912** [comment]
I've updated the file to adhere to the ESLint standards and have also added the version information to raw.version. I'm uncertain as to why it triggered another merge of the branch. I hope this won't cause any issues.

**Pull request merged**